### PR TITLE
Store and use message IDs for proper threading

### PR DESCRIPTION
**Pull request by banga**

### CODE REVIEW

File: app/bettermail.go [hunk 1]
```diff
@@ -274,6 +285,9 @@ func handleCommitCommentPayload(payload CommitCommentPayload, c context.Context)
 		SenderUserName: senderUserName,
 		Subject: subject,
 		HTMLBody: mailHtml.String(),
+		Headers: map [string]string {
+			"In-Reply-To": messageId,
```
Comment (mihaip) [inline app/bettermail.go:289]: Should this check to see if `messageId` is non-nil/empty, to handle datastore entries that were created before this change?

Comment (banga) [inline app/bettermail.go:289]: Yeah. It seems to work fine in gmail even if this header is empty, but who knows what mail client it might break.

Comment (mihaip) [inline app/bettermail.go:289]: Deployed. We'll see what breaks.

Comment (banga) [inline app/bettermail.go:289]: Ah I was just about to push the fix. Pushed it in #5

**Pull request merged**